templates/flavors/rke2/rke2ConfigTemplate.yaml

```diff
@@ -12,7 +12,12 @@ spec:
         kubelet:
           extraArgs:
             - "provider-id=linode://{{ ds.meta_data.id }}"
+      # TODO: use MDS to get public and private IP instead because hostname ordering can't always be assumed
       preRKE2Commands:
+        - |
+          mkdir -p /etc/rancher/rke2/config.yaml.d/
+          echo "node-ip: $(hostname -I | awk '{print $2}')" >> /etc/rancher/rke2/config.yaml.d/capi-config.yaml
+          sed -i "/tls-san:/a - $(hostname -I | awk '{print $1}')" /etc/rancher/rke2/config.yaml
```

Comment:
I am pretty worried about the reliance here on the ordering from `hostname -I` I think we should do this only as the absolutely last option

Reply:
The other option is a gnarly regex or curling some endpoint that can return the public IP

Reply:
I was able to start a cluster and get logs successfully with just setting the `node-ip` to the internal IP, are there other places where I could look for errors requiring the external IP to be set in the SAN as well?

Reply:
added, thanks 👍 We were seeing errors when trying to view container logs so if both `kubectl logs` works and an HA control plane is able to form we should be good. This seems to be the case with the latest commit.